libecole/src/observation/strongbranchingscores.cpp

```diff
@@ -0,0 +1,102 @@
+#include <cmath>
+#include <cstddef>
+
+#include "ecole/observation/strongbranchingscores.hpp"
+#include "ecole/scip/type.hpp"
+
+#include <scip/scip.h>
+#include <scip/scipdefplugins.h>
+#include <scip/struct_branch.h>
+#include <scip/struct_lp.h>
+#include <scip/struct_scip.h>
+#include <scip/struct_var.h>
+#include <scip/utils.hpp>
+
+namespace ecole {
+namespace observation {
+
+static auto extract_strong_branching_scores(scip::Model const& model) {
+
+	/* store original SCIP parameters */
+	bool integralcands = model.get_param_explicit<scip::ParamType::Bool>(
+		"branching/vanillafullstrong/integralcands");
+	bool scoreall = model.get_param_explicit<scip::ParamType::Bool>(
+		"branching/vanillafullstrong/scoreall");
+	bool collectscores = model.get_param_explicit<scip::ParamType::Bool>(
+		"branching/vanillafullstrong/collectscores");
+	bool donotbranch = model.get_param_explicit<scip::ParamType::Bool>(
+		"branching/vanillafullstrong/donotbranch");
+	bool idempotent = model.get_param_explicit<scip::ParamType::Bool>(
+		"branching/vanillafullstrong/idempotent");
+
+	/* set parameters for vanilla full strong branching  */
+	SCIP* scip = model.get_scip_ptr();
+
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/integralcands", true);
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/scoreall", true);
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/collectscores", true);
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/donotbranch", true);
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/idempotent", true);
+
+	/* execute vanilla full strong branching */
+	SCIP_BRANCHRULE* branchrule = SCIPfindBranchrule(scip, "vanillafullstrong");
+	SCIP_Bool allowaddcons = TRUE;
```

Comment:
This has no effect anyway, but this guy should be set to FALSE, since ideally the call to the branching rule is idempotent.